src/mrvi/_model.py

```diff
@@ -743,6 +753,111 @@ def get_aggregated_posterior(
             dist.Normal(qu_loc, qu_scale),
         )
 
+    def differential_abundance(
+        self,
+        adata: AnnData | None = None,
+        sample_cov_keys: list[str] | None = None,
+        sample_subset: list[str] | None = None,
+        minibatch_size: int = 128,
+    ) -> xr.Dataset:
+        """
+        Compute the differential abundance between samples.
+
+        Computes the logarithm of the ratio of the probabilities of each sample
+        conditioned on the estimated aggregate posterior distribution of each cell.
+
+        Parameters
+        ----------
+        adata
+            The data object to compute the differential abundance for.
+            If not given, the data object stored in the model is used.
+        sample_cov_keys
+            Keys for covariates (batch, etc.) that should also be taken into account
+            when computing the differential abundance.
+        sample_subset
+            Only compute differential abundance for these sample labels.
+        minibatch_size
+            Minibatch size for computing the differential abundance.
+
+        Returns
+        -------
+        :class:`xarray.Dataset`
+            Returns an xarray.Dataset with data variables:
+            - `log_probs`: Array of shape (n_cells, n_samples) containing the log probabilities for each cell across samples.
+            - `{cov_key}_log_probs`: For each key in `sample_cov_keys`, an array of shape (n_cells, n_cov_values) containing the log probabilities for each cell across covariate values.
+        """
+        adata = self._validate_anndata(adata)
+        us = self.get_latent_representation(adata, use_mean=True, give_z=False)
+
+        log_probs = []
+        unique_samples = adata.obs[self.sample_key].unique()
+        for sample_name in tqdm(unique_samples):
+            ap = self.get_aggregated_posterior(adata=adata, sample=sample_name)
+            n_splits = adata.n_obs // minibatch_size
+            log_probs_ = []
+            for u_rep in np.array_split(us, n_splits):
```

Comment:
Will this error if `n_obs < minibatch_size`?

Reply:
yes, good catch